salt_resources/states/basebox/miniconda.sls

```diff
@@ -0,0 +1,9 @@
+download_miniconda_installer:
+  cmd.run:
+    - name: wget http://repo.continuum.io/miniconda/Miniconda-latest-Linux-x86_64.sh -O /home/vagrant/miniconda.sh
+    - unless: test -f /usr/local/bin/miniconda/bin/conda
+
+ensure_miniconda_is_installed:
+  cmd.run:
+    - name: bash /home/vagrant/miniconda.sh -b -p /usr/local/bin/miniconda
```

Comment:
I am trying this. 

{% if not salt['file.search']('/home/vagrant/.bashrc', 'export PATH=/usr/local/bin/miniconda/bin:$PATH') %}
add_conda_to_bashrc:
  file.append:
      - name: /home/vagrant/.bashrc
      - text: 'export PATH=/usr/local/bin/miniconda/bin:$PATH'
      - runas: vagrant

export_conda_to_path:
  cmd.run:
    - name: 'export PATH=/usr/local/bin/miniconda/bin:$PATH'
    - runas: vagrant
{% endif %}

miniconda_env_install:
  cmd.run:
    - name: conda create -y --name miniconda_env pip
    - runas: vagrant